### DISCUSSION TITLE
Model Has Role with another variable

### DISCUSSION THREAD

**Discussion by samuel-lujan** [spatie/laravel-permission#2443] [Q&A]
Today my project supports users register, and user can manage, invoices, companies, units and others models. But one user has an role and this give permissions for every companies. I want to set up the role, by user and by company.

for example: my name is samuel. in the company_id 1 I`m admin but in company 2 I'm only a user. how can is setup this case, if I create a new field in the table by migration and add this on method $user->assignRole($role, $company_id) it will work?

**Comment by erikn69** [comment]
[basic-usage/teams-permissions](https://spatie.be/docs/laravel-permission/v5/basic-usage/teams-permissions)
https://github.com/spatie/laravel-permission/pull/1804
https://github.com/spatie/laravel-permission/discussions/2075

**Reply by samuel-lujan** [reply]
Pretty good my friend! Thanks :) 🚀
